packages/babel-cli/src/babel/options.js

```diff
@@ -281,3 +287,24 @@ function collect(value, previousValue): Array<string> {
 
   return previousValue ? previousValue.concat(values) : values;
 }
+
+function getOutDir(): string {
+  const outDirOptions = ["lib", "dist"];
+
+  // check package.json main field if it exists
+  if (fs.existsSync("package.json")) {
+    const packageJson = JSON.parse(fs.readFileSync("package.json"));
+
+    if (packageJson && packageJson.main) {
+      return path.dirname(packageJson.main);
+    }
+  }
+
+  outDirOptions.some(function(outDir) {
```

Comment:
I think skipping the `lib` and `dist` and only using `package.json` might be reasonable, given the concern about `lib`, but it definitely has risks either way.